File: solutions/go/binary-search/1/binary_search.go
package binarysearch

import "sort"

// SearchInts returns the index of the key in the list, -1 if it doesn't exist.
func SearchInts(list []int, key int) int {
	sort.Ints(list)

	left, right := 0, len(list)-1

	for left <= right {
		mid := (right + left) / 2

		if list[mid] > key {
			right = mid - 1
		} else if list[mid] < key {
			left = mid + 1
		} else {
			return mid
		}
	}

	return -1
}

File: solutions/python/binary-search/1/binary_search.py
def find(search_list, value):
    left, right = 0, len(search_list) - 1

    while left <= right:
        mid = (left + right) // 2

        if search_list[mid] == value:
            return mid
        
        if search_list[mid] > value:
            right = mid - 1
        else:
            left = mid + 1

    raise ValueError("value not in array")
